tests/std/tests/P0608R3_sane_variant_converting_constructor/test.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Also tests for P1957R2: Converting from T* to bool should be considered narrowing
+
+#include <assert.h>
+#include <bitset>
+#include <functional>
+#include <optional>
+#include <string>
+#include <type_traits>
+#include <variant>
+#include <vector>
+
+using namespace std;
+
+struct double_double {
+    double_double(double x) : x_(x) {}
+
+    double x_;
+};
+struct convertible_bool {
+    convertible_bool(bool x) : x_(x) {}
+    ~convertible_bool() = default;
+
+    operator bool() const noexcept {
+        return x_;
+    }
+
+    bool x_;
+};
+
+// P0608R3 examples
+static_assert(is_constructible_v<variant<string, bool>, const char*>);
+static_assert(is_constructible_v<variant<string, bool>, string>);
+static_assert(is_constructible_v<variant<char, optional<char16_t>>, char16_t>);
+static_assert(is_constructible_v<variant<int, reference_wrapper<double>>, double&>);
+static_assert(is_constructible_v<variant<float, int>, char>);
+#ifndef __EDG__ // TRANSITION, ...
+static_assert(is_constructible_v<variant<float, long int>, int>);

```

Comment:
Style: We conventionally say `long` and `long long`, never `long int` and `long long int`. Occurs below.

(There may be a few exceptions in the variant/optional/any tests whose sources were derived from libcxx's tests, but we are consistent in all other test and product code.)